packages/logger/src/correlationId.ts

```diff
@@ -0,0 +1,23 @@
+import type { JSONObject } from '@aws-lambda-powertools/commons/types';
+import { search as JMESPathSearch } from '@aws-lambda-powertools/jmespath';
+import { PowertoolsFunctions } from '@aws-lambda-powertools/jmespath/functions';
+
+const search = (expression: string, data: JSONObject) => {
```

Comment:
Please add a docstring to this function to explain usage

Reply:
Unsure because there's a lot of review comments, but `data` here could probably be `unknown` and then casted within the usage.

In a future PR, we might also want to relax the type of `search` in `@aws-lambda-powertools/jmespath` but not now.